src/pages/frames.js

```diff
@@ -0,0 +1,264 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable array-callback-return */
+/* eslint-disable consistent-return */
+/* eslint-disable no-plusplus */
+/* eslint-disable camelcase */
+
+import '../../styles/style.css';
+import '../../styles/actors.css';
+import '../../styles/quizResult.css';
+import '../../styles/movieFrameStyling.css';
+
+const actorsData = [
+  {
+    question: 'What movie is this frame from?',
+    a: 'Inception',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/inception.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'Interstellar',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/interstellar.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'Rango',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/rango.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'Saturday Night Fever',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/saturdayNightFever.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'The Godfather',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/theGodfather.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'The Hangover',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/theHangover.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'The Matrix',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/theMatrix.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'Thor: Ragnarok',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/thorRagnarok.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'Uncut Gems',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/UncutGems.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'WALL·E',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/wallE.jpg',
+  },
+  {
+    question: 'What movie is this frame from?',
+    a: 'The Wolf of Wall Street',
+    b: '',
+    c: '',
+    d: '',
+    correct: 'a',
+    imgSource: '../assets/images/movieFrames/WallStWolf.jpg',
+  },
+];
+
+// Array that will store invalid movie frame answers
+let wrongAnswers = [];
+
+async function getDataFromDb(pageNumber) {
```

Comment:
Wykonywanie tych zapytań po stronie klienta przy każdym odpaleniu quizu ma kilka problemów:
1. Znacznie opóźniamy możliwość rozpoczęcia quizu, dane muszą się załadować zanim zaczniemy.
2. Powodzenie odpalenia quizu jest uzależnione od zewnętrznego serwisu, nad którym nie mamy kontroli (SPOF).
3. Przy dużym powodzeniu naszej aplikacji wpędzamy się w niepotrzebne koszty, odpytując o te same dane przy każdym wejściu w quiz.

W tym konkretnym przypadku lepsze wydaje się pobranie tych danych tylko raz i umieszczenie listy powiedzmy 100 nazw filmów pod jakąś zmienną, bez łączenia się z API przy każdym wejściu.

Reply:
Jasne,
Proponuję takie rozwiązanie:

1. Przerzucę fetch z api filmowego do osobnego pliku JS
   - w przypadku powodzenia zmienna wrongAnswers będzie zawierała tytuły niepoprawnych filmów
   - w przypadku niepowodzenia zmienna wrongAnswers zostanie pusta
2. Export tej zmiennej 
3. Import zmienna wrongAnswers w pliku frames.js
4. Jeżeli zmienna wrongAnswers:
   - nie będzie pusta (fetch się udał) generuję quiz i pytania tak jak w kodzie powyżej
   - będzie pusta (fetch rzucił error) wygeneruję alert, że ten quiz nie był osiągalny z powodu problemu z serwerem i prośbę o wybranie innego :)



Mam nadzieję, że to dobry sposób na rozwiązanie problemu.